### PR TITLE
Ensure Zipkin V2 API requests are forwarded to V2 downstream API.

### LINKED ISSUE #45: Proxy sends v2 JSON payloads to v1 downstream API
In PR #42, support for Zipkin's V2 API was added – thanks again for that, @fbcbarbosa and @nathanleclaire!

However, I noticed that the Jaeger instance that we're currently running downstream of the honeycomb proxy started storing my services' traces under `unknown-service-name` as soon as I switched our Opentracing reporters to use the Zipkin V2 API endpoint and JSON format.

After some digging, I found out that the reason for this seems to be that the Honeycomb proxy sends these V2 JSON payloads to the V1 API endpoint, because that's what's hardcoded in [main.go](https://github.com/honeycombio/honeycomb-opentracing-proxy/blob/master/main.go#L77). That totally confuses Jaeger.

I'm happy to send in a PR for this myself, I've mainly created this issue to keep track of the problem.

### PR DESCRIPTION
**Pull request by tobiasgies**
Fixes #45.

May I ask for a thorough review from maintainers on this one? I'm not particularly proficient in golang.

### CODE REVIEW

File: app/testdata/payload_1.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[
```
Comment (tobiasgies) [inline app/testdata/payload_1.json]: Sample payload obtained from https://zipkin.io/zipkin-api/#/default/post_spans

**Comment by nathanleclaire** [comment]
ping @paulosman

**Comment by tobiasgies** [comment]
The build fails, but according to what I see on circleci, that only seems to be because of a missing watch token and also seems to affect other PRs. So I *think* this is ready for review?

**Comment by akvanhar** [comment]
Hi @tobiasgies, we have an all day meeting today, but this is top of my queue to get reviewed. The tests failing due to the missing watch token is expected, and if all other tests pass you should be all set there.

**Comment by tobiasgies** [comment]
@akvanhar Awesome, thanks! Don't rush on my behalf, but if there are any open questions, I'm also on the pollinators slack – please feel free to reach out any time.

File: app/app.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ func (a *App) handleSpansV1(w http.ResponseWriter, r *http.Request) {
 	contentType := r.Header.Get("Content-Type")
 
 	if a.Mirror != nil {
-		if err := a.Mirror.Send(payload{ContentType: contentType, Body: data}); err != nil {
+		if err := a.Mirror.Send(payload{Endpoint: "/api/v1/spans", ContentType: contentType, Body: data}); err != nil {
```
Comment (akvanhar) [inline app/app.go]: I'd recommend making "/api/v1/spans" a const at the top of the file.

File: app/app.go [hunk 3]
```diff
@@ -90,7 +90,7 @@ func (a *App) handleSpansV2(w http.ResponseWriter, r *http.Request) {
 	contentType := r.Header.Get("Content-Type")
 
 	if a.Mirror != nil {
-		if err := a.Mirror.Send(payload{ContentType: contentType, Body: data}); err != nil {
+		if err := a.Mirror.Send(payload{Endpoint: "/api/v2/spans", ContentType: contentType, Body: data}); err != nil {
```
Comment (akvanhar) [inline app/app.go]: I'd recommend making "/api/v2/spans" a const at the top of the file.

**Comment by tobiasgies** [comment]
Good point on the endpoint strings, thanks a lot. Pulled them out into constants like you suggested.

**Pull request merged**